src/vulkan/pipeline.cc

```diff
@@ -251,17 +251,39 @@ Result Pipeline::AddDescriptor(const BufferCommand* buffer_command) {
   }
 
   if (desc == nullptr) {
-    auto ssbo = MakeUnique<StorageBufferDescriptor>(
-        device_, buffer_command->GetDescriptorSet(),
+    auto desc_type = buffer_command->IsSSBO() ? DescriptorType::kStorageBuffer
+                                              : DescriptorType::kUniformBuffer;
+    auto buffer_desc = MakeUnique<BufferDescriptor>(
+        desc_type, device_, buffer_command->GetDescriptorSet(),
         buffer_command->GetBinding());
-    descriptors_.push_back(std::move(ssbo));
+    descriptors_.push_back(std::move(buffer_desc));
 
     desc = descriptors_.back().get();
   }
 
-  desc->AddToSSBODataQueue(
-      buffer_command->GetDatumType().GetType(), buffer_command->GetOffset(),
-      buffer_command->GetSize(), buffer_command->GetValues());
+  if (buffer_command->IsSSBO()) {
+    if (!desc->IsStorageBuffer()) {
```

Comment:
Once the update is pulled out these two if's can be combined (same with below)

Reply:
done